Formula/llvm.rb

```diff
@@ -59,6 +59,7 @@ def install
       lldb
       mlir
       polly
+      bolt
```

Comment:
1. Doesn't this need to be restricted to Linux? Last I checked, Bolt supports only ELF files.
2. Does this install properly? It seems the install configuration is still in flux upstream: https://github.com/llvm/llvm-project/commit/90dcdc4b6e7d86cb3d5049bd766aecddd549dd7d, https://github.com/llvm/llvm-project/commit/ad8c34bc3089d847a09bb740f7a58c96073e0959, https://github.com/llvm/llvm-project/commit/f7a33090a91015836497c75f173775392ab0304d, https://github.com/llvm/llvm-project/commit/07b749800c5cd4105d49ab46be5f0a2079dd709a, https://github.com/llvm/llvm-project/commit/0bb317b7bff3d4f70bddd03a1f337c3f2a118943
3. Is this included in the builds that upstream ship? (e.g. the builds that can be downloaded [here](https://github.com/llvm/llvm-project/releases/tag/llvmorg-14.0.6), or in the [APT packages](https://apt.llvm.org))
4. Can we add a test?

Also, LLVM takes ages to build, so we'll probably just include this in #106925, if it is suitable for packaging.

Reply:
1. Yes, it only supports ELF binaries and I should've added it under Linux OS condition.
2. With Homebrew build system, I'm not sure. I have to try it.
3. No, it is not included in the official LLVM package (I've tried the latest APT package, LLVM 15).
4. You can but I don't think the optimized binary generated by `llvm-bolt` be consistent on new releases. So, tracking the test results would be bothersome.

Reply:
> 4. You can but I don't think the optimized binary generated by `llvm-bolt` be consistent on new releases. So, tracking the test results would be bothersome.

We don't need to track this that closely. We should at least check that:
1. It installed _something_
2. That something can be used and doesn't crash when you try to make basic use of it
If doing the above is still bothersome then it's definitely not a good idea to install this.

Reply:
Oh! in those cases you've mentioned, it's simple to write a test for it.

I'll try to propose changes in your llvm-15 PR.